### ISSUE TITLE
NPE for empty JSONObject

### ISSUE THREAD

**Issue by f2prateek** [amplitude/Amplitude-Android#35]
I'm running into a null pointer exception if I pass in an empty JSONObject to `logEvent`.

Reproducible in a new project with the following lines in the `onCreate` of an Activity:

```java
    Amplitude.getInstance().initialize(this, "foo");
    Amplitude.getInstance().logEvent("bar", new JSONObject());
```

From the stacktrace (see below) I'm guessing `JSONObject#names` returns null if there are no mappings in the JSONObject.
```java
    private JSONObject cloneJSONObject(final JSONObject obj) {
        JSONArray nameArray = obj.names(); // becomes null
        int len = nameArray.length();
```

```
04-03 14:38:30.396    1619-1619/com.f2prateek.myapplication E/AndroidRuntime﹕ FATAL EXCEPTION: main
    Process: com.f2prateek.myapplication, PID: 1619
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.f2prateek.myapplication/com.f2prateek.myapplication.MainActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'int org.json.JSONArray.length()' on a null object reference
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2325)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2387)
            at android.app.ActivityThread.access$800(ActivityThread.java:151)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1303)
            at android.os.Handler.dispatchMessage(Handler.java:102)
            at android.os.Looper.loop(Looper.java:135)
            at android.app.ActivityThread.main(ActivityThread.java:5254)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'int org.json.JSONArray.length()' on a null object reference
            at com.amplitude.api.AmplitudeClient.cloneJSONObject(AmplitudeClient.java:290)
            at com.amplitude.api.AmplitudeClient.logEvent(AmplitudeClient.java:186)
            at com.f2prateek.myapplication.MainActivity.onCreate(MainActivity.java:17)
            at android.app.Activity.performCreate(Activity.java:5990)
            at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1106)
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2278)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2387)
            at android.app.ActivityThread.access$800(ActivityThread.java:151)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1303)
            at android.os.Handler.dispatchMessage(Handler.java:102)
            at android.os.Looper.loop(Looper.java:135)
            at android.app.ActivityThread.main(ActivityThread.java:5254)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)

```

**Comment by f2prateek** [comment]
This will verify that `JSONObject#names` will return null.

```java
    JSONObject jsonObject = new JSONObject();
    JSONArray array = jsonObject.names();
    if (array == null) {
      throw new AssertionError("null array");
    }
```

**Comment by allanca** [comment]
Thanks for catching and reporting this crash. I've committed a fix and will release 1.5.1 with the fix shortly.
